src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,34 +27,40 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+
+        String[] splitUp = address.split(",");
+        if(splitUp.length != 4){
```

Comment:
Testing the length of `splitUp` is the simplest way to do validation. But in the future, if you need to do more sophisticated validation, you cannot write all those checking codes in one condition. So you still should use the method `isValidAddress(...)` here, and write the checking details in that method.